### PR TITLE
 PLAT-165: CoreGroup for model-level permissions  

### PR DESCRIPTION
**Pull request by docktorrr**
## Purpose
We need to implement the model-level permissions inside an Organization.

## Approach
- The idea is that there are two types of permissions: roles (OrgAdmin, WorkflowAdmin etc) and model-level permissions (to allow user to perform particular actions with particular models).
Currently we have Django built-in Groups, they are used to define Roles of the user. We need to implement model-level permissions. I added CoreGroup model for that, it is similar to Django Group, but it is associated with an organization. I left normal Groups as they are, because they do their job for defining Role permissions well.
- CoreGroup has full CRUD endpoint with permission only for Org Admin.
- Added Permissions list endpoint (it may be usefull when we want to set or remove some permissions for CoreGroup).
- Renamed ProgramAdmin and ProgramTeam roles to WorkflowAdmin and WorkflowTeam.
- Some minor refactoring

### CODE REVIEW

File: workflow/serializers.py [hunk 1]
```diff
@@ -58,6 +64,46 @@ class Meta:
...
+        return coregroup
+
+    def update(self, instance, validated_data):
+        # update permissions
+        new_permissions = validated_data.pop('permissions', [])
+        old_permissions = instance.permissions.values_list('id', flat=True)
+        to_add = set(new_permissions) - set(old_permissions)
+        to_delete = set(old_permissions) - set(new_permissions)
+        instance.permissions.remove(*to_delete)
+        instance.permissions.add(*to_add)
```
Comment (jefmoura) [inline workflow/serializers.py]: You don't need this logic to remove old permissions and add new ones, you can just use the command `set()`
https://docs.djangoproject.com/en/1.11/topics/db/examples/many_to_many/

File: workflow/views.py [hunk 2]
```diff
@@ -62,17 +62,32 @@ class DefaultCursorPagination(CursorPagination):
...
+
+    retrieve:
+    Return the given permission
+
+    list:
+    Return a list of existing permissions
+    """
+    queryset = Permission.objects.all()
+    serializer_class = serializers.PermissionSerializer
+    permission_classes = (IsOrgMember,)
```
Comment (jefmoura) [inline workflow/views.py]: Permissions don't have a relationship with an organization, so why did you add this permission class?

**Review by jefmoura — CHANGES_REQUESTED** [review]

File: workflow/serializers.py (hunk 1 above)
Comment (docktorrr) [reply workflow/serializers.py]: Yes, thanks for the hint, stuck in the old django versions.

File: workflow/views.py (hunk 2 above)
Comment (docktorrr) [reply workflow/views.py]: Changed to AllowAuthenticatedRead (just not to let everyone to see them)

**Review by jefmoura — APPROVED** [review]
